src/events/messageReactionAdd.ts

```diff
@@ -1,15 +1,28 @@
+import { ThreadChannel } from 'discord.js';
 import { event } from 'jellycommands';
+import { MESSAGE_READ, REACTION_ROLE_CHANNEL } from '../config'
 import { hasPermission } from '../utils/reactionHandler';
 
 export default event({
     name: 'messageReactionAdd',
     run: async (_, reaction, user) => {
         try {
-            const result = await hasPermission(reaction, user);
-
-            result.member.roles.add(result.roleId);
+            // The bot shouldn't react to its own reactions
+            if (user.bot)
+                return
+            // If this is a reaction in the role reaction channel
+            if (reaction.message.channelId === REACTION_ROLE_CHANNEL) {
+                const result = await hasPermission(reaction, user);
+                result.member.roles.add(result.roleId);
+            } else {
+                // Otherwise we make a more general guess that if the original message was from the bot
+                // and it's a MESSAGE_READ emoji reaction, we assume the message should be deleted
+                if (reaction.message.author.bot && reaction.emoji.name === MESSAGE_READ) {
+                    reaction.message.delete()
```

Comment:
I'm wondering if being a little more restrictive and also making sure the message is in a channel where we want this to happen would be safer? An additional check in the if statement

Reply:
Imo it's fine since we already ensure that it's the bots own message that it deletes, so the only scenario where this would be too permissive is if the bot creates a message somewhere, with this exact reaction to it, and the intention is not for the message to be deleted when the reaction is added. It's fairly edge case. But I'm adding a check to ensure it's one of the SOLVABLE_FORUMS, we can set up a different list of channels that it should be capable of deleting in later if it becomes relevant.

Reply:
Cool. Figured it wouldn't be an issue right now in the current state but could be something that's a nightmare to troubleshoot what's going on in the future if it pops up